uxarray/core/dataarray.py

```diff
@@ -1104,3 +1103,116 @@ def _slice_from_grid(self, sliced_grid):
             dims=self.dims,
             attrs=self.attrs,
         )
+
+    def neighborhood_filter(
```

Comment:
This implementation looks great! May we move the bulk of the logic into the `uxarray.grid.neighbors` module and call that helper from here?

We can keep the data-mapping checks here, and anything related to constructing and returining the final data array but the bulk of the computations would go inside a helper in the module mentioned above.

Reply:
I wouldn't know how to implement that, but I am happy to defer to you.